src/wp-includes/blocks.php

```diff
@@ -201,6 +201,15 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	}
 	$metadata['file'] = $metadata_file;
 
+	/**
+	 * Filters the metadata provided for registering a block type.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param array $metadata Metadata for registering a block type.
+	 */
+	$metadata = apply_filters( 'pre_register_block_type_from_metadata', $metadata );
```

Comment:
`pre_` filters in WP are usually meant for short-circuiting logic within a function.

I'd change the naming here.

Maybe just `register_block_type_from_metadata` or `block_type_metadata`